src/core/image_resizer.js

```diff
@@ -268,6 +279,90 @@ class ImageResizer {
     return imgData;
   }
 
+  _rescaleImageData() {
+    const { _imgData: imgData } = this;
+    const { data, width, height, kind } = imgData;
+    // K is such as width * height * 4 / 2 ** K <= 2 ** 31 - 1
+    const K = Math.ceil(Math.log2((width * height * 4) / (2 ** 31 - 1)));
```

Comment:
Following the suggestion above.
```suggestion
    // K is such as width * height * 4 / 2 ** K <= 2 ** 31 - 1
    const K = Math.ceil(Math.log2((width * height * 4) / MAX_INT_32));
```